### PR TITLE
Update submission `/edit` endpoint to `/enketo/<action>` and accept `view` or `edit`

### LINKED ISSUE #2992: Update submission `/edit` endpoint for view-only change in kobocat
From the community request [here](https://community.kobotoolbox.org/t/ability-to-view-completed-submissions-in-enketo-and-ideally-pdf-them/15457/7), some changes are required for `kpi` to handle the update from `kobocat` [here](https://github.com/kobotoolbox/kobocat/pull/672). The current submission's `/edit` endpoint must be updated to `/enketo/<action>/` where `<action>` can be either `view` or `edit`.

### PR DESCRIPTION
**Pull request by joshuaberetta**
Replace submission `/edit` endpoint with more versatile `/enketo/<action>` endpoint to allow for either `edit` or `view` actions. A `GET` request to `/enketo/edit` will return the same response as the previous `/edit` endpoint and a `GET` request to `/enketo/view` will return a view-only URL for a submission in the Enketo UI.

If the user has `change_submissions` permissions, the submission modal will be:

Screen Shot 2021-06-23 at 4 14 03 PM

If the user only has `view_submissions` permissions, the modal will be:

Screen Shot 2021-06-23 at 4 11 27 PM

## Description

Allow for view-only access to submissions in Enketo UI.

## Related issues

closes #2992 
part of https://github.com/kobotoolbox/kobocat/issues/671
blocked by https://github.com/kobotoolbox/kobocat/pull/672
blocked by #3322

### CODE REVIEW

File: jsapp/js/dataInterface.es6 [hunk 1]
```diff
@@ -633,7 +633,13 @@ export var dataInterface;
     },
     getEnketoEditLink(uid, sid) {
       return $ajax({
-        url: `${ROOT_URL}/api/v2/assets/${uid}/data/${sid}/edit/?return_url=false`,
+        url: `${ROOT_URL}/api/v2/assets/${uid}/data/${sid}/enketo/edit/?return_url=false`,
+        method: 'GET'
+      });
+    },
+    getEnketoViewLink(uid, sid) {
```
Comment (noliveleger) [inline jsapp/js/dataInterface.es6:640]: Where can we use this call in the UI?

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: jsapp/js/components/submissions/submissionModal.es6 [hunk 2]
```diff
@@ -169,12 +170,21 @@ class SubmissionModal extends React.Component {
       isEditLoading: true,
       isEditingDuplicate: true,
     });
-    enketoHandler.editSubmission(this.props.asset.uid, this.state.sid).then(
+    enketoHandler.openSubmission(this.props.asset.uid, this.state.sid, 'edit').then(
```
Comment (magicznyleszek) [inline jsapp/js/components/submissions/submissionModal.es6:173]: Please make `'edit'` and `'view'` a constant defined in `enketoHandler`. I wanted to describe here our way of defining enum-like objects, but ended up improving our ways and there is new PR: kpi#3321. I don't know how much more work you have here, but if the `createEnum` PR is merged first, please use it :)

File: jsapp/js/components/submissions/submissionModal.es6 [hunk 3]
```diff
@@ -476,6 +486,16 @@ class SubmissionModal extends React.Component {
                 </a>
               }
 
+              {this.userCan('view_submissions', this.props.asset) &&
+                <a
+                  onClick={this.launchViewSubmission.bind(this)}
+                  className='kobo-button kobo-button--blue submission-duplicate__button'
```
Comment (magicznyleszek) [inline jsapp/js/components/submissions/submissionModal.es6:492]: I think when you click it, it should be disabled while loading, so `disabled={this.state.isViewLoading}` :)

File: jsapp/js/enketoHandler.es6 [hunk 4]
```diff
@@ -45,31 +44,36 @@ const enketoHandler = {
...
    * @returns {Promise} Promise that resolves when url is being opened.
    */
-  editSubmission(aid, sid) {
+  openSubmission(aid, sid, action) {
     // we create the tab immediately to avoid browser popup blocker killing it
     this.winTab = window.open('', '_blank');
-    const editPromise = new Promise((resolve, reject) => {
-      if (this._hasEnketoUrl(aid, sid)) {
-        this._openEnketoUrl(aid, sid);
+    let dataInt = dataInterface.getEnketoEditLink;
```
Comment (magicznyleszek) [inline jsapp/js/enketoHandler.es6:50]: Please use a bit more descriptive variable name - I thought that `dataInt` is a variable that is a shortcut to `dataInterface`. Maybe `dataIntMethod`?

**Review by magicznyleszek — CHANGES_REQUESTED** [review]

**Review by magicznyleszek — APPROVED** [review]

**Review by noliveleger — APPROVED** [review]

**Comment by ahmad1k** [comment]
Hi, can I get these links? I mean previously I was using " https://kc.humanitarianresponse.info/api/v1/data" link
Thanks

**Comment by joshuaberetta** [comment]
Hi @ahmad1k, this feature has not yet been released into production.

**Pull request merged**